clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -400,12 +412,33 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
             int totalBytes = 0;
             int totalCount = 0;
             FetchResponse response = new FetchResponse(resp.responseBody());
+
+            // Look for OffsetOutOfRange error in fetchResponse
+            Map<TopicPartition, Long> offsetOutOfRangePartitions = new HashMap();
+            for (Map.Entry<TopicPartition, FetchResponse.PartitionData> entry : response.responseData().entrySet()) {
```

Comment:
Then we may have the following sequence of events:

1) consumer.poll() successfully returns data for partition 1 and 2
2) consumer.seek() gives partition 1 and invalid position
3) consumer.poll() throws OffsetOutOfRange exception 
4) consumer.seek() to another invalid position for partition 1
5) consumer.poll() returns data for partition 2
6) consumer.poll() throws OffsetOutOfRange exception
...

It appears a bit counter intuitive to me that the 2nd poll() after the seek() caused the OffsetOutOfRange exception. In short it adds a little bit code complexity for a little bit performance improvement. I still slightly prefer the current solution though. I will do it if you are strong on your solution.